OpenFOAM CalculiX XY (issue#34)

Alphaoo1 opened the pull request:
A flap tutorial case for OpenFOAM-CalculiX, in the XY direction as needed in #34.
The meshes are created using Salome.

If some files are deleted during `./Allclean`, Readme.md contains instructions to recreate those files.
preCICE version 2.0.2
OpenFOAM v7
CalculiX v2.15

Fixes #34.

MakisH commented:
Thank you very very much, @Alphaoo!

I think it would be a good idea to actually change the files of the current tutorial, since this is a better default. For people that have already built their cases based on the current orientation, this PR would be a clear indication of what they could change. Moreover, it would make reviewing easier, as we could see the changes only. If anybody still wants to use the previous case, it is always available on GitHub.

@uekerman I vote for replacing the current case instead of adding a new one. What do you think?

@Eder-K this will also then need an update in the reference results of the system tests.

uekerman commented:
> @uekerman I vote for replacing the current case instead of adding a new one. What do you think?

Yes, replacing is better than a new one. Is this only for `flap_perp/OpenFOAM-CalculiX`? What about the other cases 3D-3D and 3D-2D cases? Isn't this a larger problem?

MakisH commented:
Yes, this affects also the other cases and we should gradually port them as well. For deal.II it will be easy, for SU2 we would need help (maybe a nice first contribution for anyone interested in https://github.com/precice/su2-adapter/issues/16).

Alphaoo1 commented:
I have added the [OpenFOAM+CalculiX CHT](https://precice.discourse.group/t/precice-mesh-id-1-error-openfoam-calculix-cht/313/7) case, which have some issues from CalculiX side.
I took the general setup of the flap case from the FSI.

I have adapted the CHT part similar to the heat-exchanger case.

I am currently working on making a flow over the heat plate case similar to [OF+Nutils](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-nutils). I will push as soon as I am done, working towards Issue #103 .

MakisH commented:
> I have added the [OpenFOAM+CalculiX CHT](https://precice.discourse.group/t/precice-mesh-id-1-error-openfoam-calculix-cht/313/7) case, which have some issues from CalculiX side.
> I took the general setup of the flap case from the FSI.
> 
> I have adapted the CHT part similar to the heat-exchanger case.
> 
> I am currently working on making a flow over the heat plate case similar to [OF+Nutils](https://github.com/precice/tutorials/tree/master/CHT/flow-over-plate/buoyantPimpleFoam-nutils). I will push as soon as I am done, working towards Issue #103 .

Thank you very much for contributing, @Alphaoo1! Could you please split this PR to a different one for each tutorial? This would make the review much easier. Otherwise, this may stay an open PR for too long.

Alphaoo1 commented:
>Could you please split this PR to a different one for each tutorial? This would make the review much easier. Otherwise, this may stay an open PR for too long.

@MakisH Yes. No Problem.

MakisH commented:
@Alphaoo1 could you please also remove all the result files? You can keep a screenshot in the README to help users understand what this is about, but currently this PR has way too many files to start reviewing it. You can have a look at the [files changed tab](https://github.com/precice/tutorials/pull/92/files), where I would like to be able to browse the code and add comments.

Thank you for keeping this up to date all this time! :smiley:

Alphaoo1 commented:
@MakisH I have tried to clean up all the things I could think of.

You might find that there are copies of `PolyMesh`, it is because when we do clean case OpenFOAM clears everything, along with the `PolyMesh` directory. 

Let me know if I need to do something else. :smile:

MakisH commented on CHT/flow-over-plate/buoyantPimpleFoam-nutils/OpenFOAM/Extrusion_1.stl:
```diff
@@ -0,0 +1,86 @@
+solid 
```
I guess we can remove all the new files in `CHT/flow-over-plate/buoyantPimpleFoam-nutils/`.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Allrun:
```diff
@@ -68,7 +68,7 @@ Solver2="ccx_preCICE"
         # We use CalculiX CGX to setup the structural simulation
         echo "  Executing cgx (provided by CalculiX, make sure this exists)..."
-        cgx -bg pre_flap.fbd > prepare_flap.log 2>&1
+#         cgx -bg pre_flap.fbd > prepare_flap.log 2>&1
```
This has now been removed in #95. Please merge.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Allclean:
```diff
@@ -12,6 +12,9 @@ cd ${Participant1}
     # Clean the case
     cleanCase
     rm -rfv 0
+    cd constant
+        cp -rfv polyMesh_backup polyMesh
+    cd ..
```
See detailed comment in the conversation: let's not copy an already created mesh, but add the steps to rotate the created mesh into the run scripts.

Alphaoo1 replied:
> Thank you, @Alphaoo1! I additionally removed all the `OpenFOAM-CalculiX_interFoam/Solid/*.vtu` files and we are now down to 88 files: that's more manageable. 
>
Thank you for your patience @MakisH, I am a bit new to git workflow so not yet up to the standards.

> > You might find that there are copies of PolyMesh, it is because when we do clean case OpenFOAM clears everything, along with the PolyMesh directory.
> 
> I see that you are essentially taking the already produced mesh and rotating it. This is now an ideal solution, however, because:
> 
> 1. we need to distribute the mesh, which adds several new files and significantly increases the size of the repository.
> 2. the user cannot easily modify the (already produced) mesh.
> 

No I am not rotating the already produced mesh.
I use `ideasUnvToFoam Fluid_xy_original.unv` to create the `polyMesh` directory.
I tried rotating the mesh using the commands available in OpenFOAM, it works fine, but the rotation of CalculiX is a bit tricky and it does not exactly match with Fluid mesh. This is why I chose to make a new mesh all together.

> I understand also that this is the easy solution and manually converting the blockMeshDict would be painful. I would then propose the following: let's keep the original blockMeshDict files and add the rotation commands in the run scripts. Then the user should be happy, even though a bit puzzled for the additional step. But this is then future problem.
> 
As I said earlier, it's a bit more complicated to directly rotate mesh and have it aligned with Solid mesh.
I went through the files again, and we only need to keep the `boundary` file from `polyMesh_backup`.
When we do `ideasUnvToFoam Fluid_xy_original.unv`, it creates a `boundary` file with all the boundaries with `type patch`. We have `type wall` for some boundaries, so I chose to copy the polyMesh folder.
I have added this command to `runFluid` if we want to go in this direction.

> Before we start looking into details, let's also break this PR to smaller ones, one for each tutorial. With a [quick look](https://github.com/precice/tutorials/pull/92/files), I see changes/additions in the following directories:
> 
> * `CHT/flow-over-plate/buoyantPimpleFoam-nutils/`: I assume these are probably residues from a previous run, so we can remove all the new files. We anyway don't need to rotate this case.
Yes. Will do

> * `FSI/flap_perp/OpenFOAM-CalculiX`: These changes stay in this PR.
>   
>   * I see also a binary file `FSI/flap_perp/OpenFOAM-CalculiX/Fluid/Fluid_XY.hdf`. What is this? We probably don't need it here. Similarly for `FSI/flap_perp/OpenFOAM-CalculiX/Fluid/Fluid_xy_original.unv `.
>   * I see that you renamed the `blockMeshDict` to `blockMeshDict_original`. Let's rename it back to `blockMeshDict` and use it.
The `.hdf` file is the file for original geometry and mesh created in Salome. `.unv` file is exported from Salome and converted into `polyMesh` using the command I mentioned earlier.

> * ` FSI/flap_perp/OpenFOAM-CalculiX_interFoam`: This is a different case, let's move it to a separate PR. Many points of the review will also apply there, so let's first get this one merged.
> 
Yes, I will create a new PR for the interFoam.

> In any case, please make your new PRs to target the `develop` branch of the tutorials. You can also change the target branch of this one, by clicking `edit` next to the title. Preferably merge the latest updates from [develop](https://github.com/precice/tutorials/tree/develop) into your branch first (`git merge develop`). Everything will hopefully be merged automatically, without any conflicts.

Ok. I will do the changes.
Let me know what do you think about my comments. and I will make changes accordingly.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Allclean:
```diff
@@ -29,8 +32,8 @@ rm -fv ${Participant1}_reconstructPar.log
     # Clean the case
-	rm -fv *.nam
-	rm -fv *.msh
+# 	rm -fv *.nam
+# 	rm -fv *.msh
```
Why keeping these files? Similarly to the OpenFOAM files, let's add the rotation commands to the run scripts.

Alphaoo1 replied:
>Why keeping these files? 
As we have `pre_flap.fbd` in the original tutorial case, it has commands to export the `.nam` and `.msh` files. But in this case, I had to do it manually. The commands are the same but I haven't yet figured out how to write a script to export all these files.

>Similarly to the OpenFOAM files, let's add the rotation commands to the run scripts.
CalculiX does not rotate around the same point as OpenFOAM, and ultimately we have a bit mismatching meshes.
(one does rotation around the origin and the other does rotation around a corner, but I don't remember now which is which).
It was easier for me to just create new geometry and new mesh in Salome. :sweat_smile: 

But yes, Salome workflow might not be fitting to everyone.

MakisH replied:
Probably the Salome workflow would be a quite pragmatic approach: many users use it anyway. However, we try to keep the dependencies of the tutorials to a minimum (preCICE already has too many dependencies and components). In any case, this is a tutorial to learn how to use preCICE and its adapters, not how to make a perfect FSI simulation.

What if we just rotate the CalculiX case by hand, e.g. using the Salome output as an example?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Allrun:
```diff
@@ -32,7 +32,7 @@ Participant1="Fluid"
 
     echo "  Preparing the mesh..."
-    blockMesh -case ${Participant1} > ${Participant1}_blockMesh.log 2>&1
+#     blockMesh -case ${Participant1} > ${Participant1}_blockMesh.log 2>&1
```
We want to keep the mesh generation with blockMesh. After this line, we also need to rotate the mesh.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Allrun_parallel:
```diff
@@ -1 +0,0 @@
-./Allrun -parallel
```
Why deleting this file?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Fluid/constant/polyMesh/boundary:
```diff
@@ -0,0 +1,62 @@
+/*--------------------------------*- C++ -*----------------------------------*\
```
After we add the rotation into the run scripts, we need to remove all the `polyMesh` directories.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Fluid/constant/transportProperties:
```diff
@@ -12,4 +12,6 @@ transportModel  Newtonian;
 
 nu              nu [ 0 2 -1 0 0 0 0 ] 1e-03;
 
+rho             rho [1 -3 0 0 0 0 0] 1e-0;
```
How is this change related to the rotation? Let's remove this change.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/README.md:
```diff
@@ -1,23 +1,44 @@
-# Tutorial for an FSI simulation of an elastic flap perpendicular to a channel flow
+## flap_perp_Salome_XY
```
Why changing the title?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/README.md:
```diff
@@ -1,23 +1,44 @@
-This tutorial is described in the [preCICE wiki](https://github.com/precice/precice/wiki/Tutorial-for-FSI-with-OpenFOAM-and-CalculiX).
 
-It is known to work with OpenFOAM 4.1, 5.0, v1712, v1806 and CalculiX 2.13 with CGX, but it should also work with newer versions. Have a look on issues concerning [OpenFOAM 6 / dev](https://github.com/precice/openfoam-adapter/issues/21) and [OpenFOAM v1812](https://github.com/precice/openfoam-adapter/issues/59).
+- In this case, all the geometries and meshes are created with Salome
```
Instead of using Salome, we can probably achieve the same using:
- [rotateMesh](https://www.openfoam.com/documentation/guides/latest/man/rotateMesh.html) for OpenFOAM
- hopefully some similar tool for CalculiX (or manually, as it is rather small)

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/README.md:
```diff
@@ -1,23 +1,44 @@
+- use [`CalculiX4Caelinux` CalculiX launcher](http://calculixforwin.blogspot.com/2015/05/calculix-launcher.html) or [`unv2ccx`](https://github.com/calculix/unv2ccx/releases)
+- Once the `.unv` file is converted. Copied the `flap.inp` file from preCICE tutorial case. In the `CalculiX4Caelinux`, open the `Mesh_name_OUT.inp` in pre-processor. 
+- While on the `cgx` window, type `prnt se`. This will list out all the group names.
+- Use following commands
```
What for? I guess that we need to discuss again about the README after we settle on the approach.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Solid/flap.inp:
```diff
@@ -1,28 +1,33 @@
+** fix on all nodes for translation in z-dir, rotation about x- and y-axis
 *BOUNDARY
-Nall, 2, 2, 0 
+Nall, 3, 3, 0 
```
Why this change?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Solid/flap.inp:
```diff
@@ -1,28 +1,33 @@
+Nall, 3, 3, 0 
 Nall, 4, 4, 0 
-Nall, 6, 6, 0 
+Nall, 5, 5, 0 
```
Why this change?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/Solid/flap.inp:
```diff
@@ -1,28 +1,33 @@
 *BOUNDARY
 Nfix1, 1, 3
-** fix on all nodes for translation in y-dir, rotation about x- and z-axis
+
```
These whitespace changes are probably done automatic by the editor. We should have some way of automatically formatting these files, but since this is quite tricky, it would be nice if you could avoid these. I can also try to cleanup these at the end.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/precice-config.xml:
```diff
@@ -30,22 +30,22 @@
     <coupling-scheme:serial-explicit>
         <time-window-size value="1e-2"/>
-        <max-time value="5"/>
+        <max-time value="1"/>
```
Nice addition, but not related to the rotation, so let's leave it like this or discuss it in a separate PR.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/runFluid:
```diff
@@ -22,7 +22,7 @@ echo "Preparing and running the Fluid participant..."
 rm -rfv Fluid/0/
 cp -r Fluid/0.orig/ Fluid/0/
-blockMesh -case Fluid
+# blockMesh -case Fluid
```
Similarly to `Allrun`: let's keep this.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX/runSolid:
```diff
@@ -11,9 +11,9 @@
-cd Solid
-    cgx -bg pre_flap.fbd
-cd -
+# cd Solid
+#     cgx -bg pre_flap.fbd
+# cd -
```
Similarly to `Allrun`: let's keep this.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX_interFoam/Allclean:
```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
```
Please move the ` FSI/flap_perp/OpenFOAM-CalculiX_interFoam/` into a separate PR.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX_interFoam/Fluid/system/fvSchemes.txt:
```diff
@@ -0,0 +1,54 @@
+/*--------------------------------*- C++ -*----------------------------------*\
```
Is this a temporary backup file?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX_interFoam/Fluid/system/fvSolution.txt:
```diff
@@ -0,0 +1,97 @@
+/*--------------------------------*- C++ -*----------------------------------*\
```
Similarly: temporary backup file?

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX_interFoam/README.md:
```diff
@@ -0,0 +1,19 @@
+# Tutorial for an FSI simulation of an elastic flap perpendicular to a channel flow
```
This README probably needs to be updated to correspond to the interFoam case.

Alphaoo1 replied:
I will work on this.

MakisH commented on FSI/flap_perp/OpenFOAM-CalculiX_interFoam/precice-config.xml:
```diff
@@ -0,0 +1,137 @@
+
+    <m2n:sockets from="Fluid" to="Calculix" />
+
+    <!--<coupling-scheme:serial-explicit>
+        <time-window-size value="1e-4"/>
+        <max-time value="1.0"/>
+        <participants first="Fluid" second="Calculix"/>
+        <exchange data="Forces0" mesh="Solid" from="Fluid" to="Calculix"/>
+        <exchange data="Displacements0" mesh="Solid" from="Calculix" to="Fluid"/>
+    </coupling-scheme:serial-explicit>-->
+
+ 
+    <!--<coupling-scheme:serial-implicit>
+            <time-window-size value="1.e-3" />
+            <max-time value="0.5"/>
+            <participants first="Fluid" second="Calculix"/>
+            <exchange data="Forces0" mesh="Solid" from="Fluid" to="Calculix"/>
+            <exchange data="Displacements0" mesh="Solid" from="Calculix" to="Fluid" />
+
+            <max-iterations value="100"/>
+            <absolute-convergence-measure limit="1e-6" data="Displacements0" mesh="Solid"/>
+            <relative-convergence-measure limit="1e-2" data="Forces0" mesh="Solid"/>
+            <extrapolation-order value="2"/>
+ 
+            <acceleration:IQN-ILS>
+                <data name="Displacements0" mesh="Solid"/>
+                <preconditioner type="residual-sum"/>
+                <filter type="QR1" limit="1e-6"/>
+                <initial-relaxation value="0.01"/> 
+                <max-used-iterations value="15"/>
+                <time-windows-reused value="0"/> 
+            </acceleration:IQN-ILS> 
+
+    </coupling-scheme:serial-implicit>-->
```
This file needs some clean-up. We should choose a config that is robust but also uses default values wherever possible.

MakisH reviewed (changes requested):
Congratulations of your first preCICE PR review! :partying_face: 

Thank you very much for all the work, please look through my comments and try to apply all the suggestions, if possible, to make this case consistent with the rest of the preCICE tutorials.

See also a separate detailed comment in the discussion, but in this review my main points are:
- Change the approach from Salome-based mesh generation to OpenFOAM/CalculiX-native rotation at preprocessing time.
- Remove some non-relevant changes.
- Move the interFoam-based tutorial to a different PR, preferably after applying the results from our discussion here.

MakisH commented:
Thank you, @Alphaoo1! I additionally removed all the `OpenFOAM-CalculiX_interFoam/Solid/*.vtu` files and we are now down to 88 files: that's more manageable. :smile: 

> You might find that there are copies of PolyMesh, it is because when we do clean case OpenFOAM clears everything, along with the PolyMesh directory.

I see that you are essentially taking the already produced mesh and rotating it. This is now an ideal solution, however, because:
1. we need to distribute the mesh, which adds several new files and significantly increases the size of the repository.
2. the user cannot easily modify the (already produced) mesh.

I understand also that this is the easy solution and manually converting the blockMeshDict would be painful. I would then propose the following: let's keep the original blockMeshDict files and add the rotation commands in the run scripts. Then the user should be happy, even though a bit puzzled for the additional step. But this is then future problem.

Before we start looking into details, let's also break this PR to smaller ones, one for each tutorial. With a [quick look](https://github.com/precice/tutorials/pull/92/files), I see changes/additions in the following directories:
- `CHT/flow-over-plate/buoyantPimpleFoam-nutils/`: I assume these are probably residues from a previous run, so we can remove all the new files. We anyway don't need to rotate this case.
- `FSI/flap_perp/OpenFOAM-CalculiX`: These changes stay in this PR.
   - I see also a binary file `FSI/flap_perp/OpenFOAM-CalculiX/Fluid/Fluid_XY.hdf`. What is this? We probably don't need it here. Similarly for `FSI/flap_perp/OpenFOAM-CalculiX/Fluid/Fluid_xy_original.unv `.
   - I see that you renamed the `blockMeshDict` to `blockMeshDict_original`. Let's rename it back to `blockMeshDict` and use it.
- ` FSI/flap_perp/OpenFOAM-CalculiX_interFoam`: This is a different case, let's move it to a separate PR. Many points of the review will also apply there, so let's first get this one merged.

In any case, please make your new PRs to target the `develop` branch of the tutorials. You can also change the target branch of this one, by clicking `edit` next to the title. Preferably merge the latest updates from [develop](https://github.com/precice/tutorials/tree/develop) into your branch first (`git merge develop`). Everything will hopefully be merged automatically, without any conflicts.

Alphaoo1 commented:
> Thank you, @Alphaoo1! I additionally removed all the `OpenFOAM-CalculiX_interFoam/Solid/*.vtu` files and we are now down to 88 files: that's more manageable. 
>

Thank you for your patience @MakisH, I am a bit new to git workflow so not yet up to the standards.

> > You might find that there are copies of PolyMesh, it is because when we do clean case OpenFOAM clears everything, along with the PolyMesh directory.
> 
> I see that you are essentially taking the already produced mesh and rotating it. This is now an ideal solution, however, because:
> 
> 1. we need to distribute the mesh, which adds several new files and significantly increases the size of the repository.
> 2. the user cannot easily modify the (already produced) mesh.
> 

No I am not rotating the already produced mesh.
I use `ideasUnvToFoam Fluid_xy_original.unv` to create the `polyMesh` directory.
I tried rotating the mesh using the commands available in OpenFOAM, it works fine, but the rotation of CalculiX is a bit tricky and it does not exactly match with Fluid mesh. This is why I chose to make a new mesh all together.

> I understand also that this is the easy solution and manually converting the blockMeshDict would be painful. I would then propose the following: let's keep the original blockMeshDict files and add the rotation commands in the run scripts. Then the user should be happy, even though a bit puzzled for the additional step. But this is then future problem.
> 

As I said earlier, it's a bit more complicated to directly rotate mesh and have it aligned with Solid mesh.
I went through the files again, and we only need to keep the `boundary` file from `polyMesh_backup`.
When we do `ideasUnvToFoam Fluid_xy_original.unv`, it creates a `boundary` file with all the boundaries with `type patch`. We have `type wall` for some boundaries, so I chose to copy the polyMesh folder.
I have added this command to `runFluid` if we want to go in this direction.

> Before we start looking into details, let's also break this PR to smaller ones, one for each tutorial. With a [quick look](https://github.com/precice/tutorials/pull/92/files), I see changes/additions in the following directories:
> 
> * `CHT/flow-over-plate/buoyantPimpleFoam-nutils/`: I assume these are probably residues from a previous run, so we can remove all the new files. We anyway don't need to rotate this case.

Yes. Will do

> * `FSI/flap_perp/OpenFOAM-CalculiX`: These changes stay in this PR.
>   
>   * I see also a binary file `FSI/flap_perp/OpenFOAM-CalculiX/Fluid/Fluid_XY.hdf`. What is this? We probably don't need it here. Similarly for `FSI/flap_perp/OpenFOAM-CalculiX/Fluid/Fluid_xy_original.unv `.
>   * I see that you renamed the `blockMeshDict` to `blockMeshDict_original`. Let's rename it back to `blockMeshDict` and use it.

The `.hdf` file is the file for original geometry and mesh created in Salome. `.unv` file is exported from Salome and converted into `polyMesh` using the command I mentioned earlier.

> * ` FSI/flap_perp/OpenFOAM-CalculiX_interFoam`: This is a different case, let's move it to a separate PR. Many points of the review will also apply there, so let's first get this one merged.
> 

Yes, I will create a new PR for the interFoam.

> In any case, please make your new PRs to target the `develop` branch of the tutorials. You can also change the target branch of this one, by clicking `edit` next to the title. Preferably merge the latest updates from [develop](https://github.com/precice/tutorials/tree/develop) into your branch first (`git merge develop`). Everything will hopefully be merged automatically, without any conflicts.

Ok. I will do the changes.
Let me know what do you think about my comments. and I will make changes accordingly.

MakisH commented:
> Thank you for your patience @MakisH, I am a bit new to git workflow so not yet up to the standards.

Welcome to the git workflow! :hugs: I am always very happy to help anybody trying to learn, especially if it results in a contribution here.

Pro tip: you are currently requesting to merge from your master branch. But now with all these changes I proposed you will mess up your current production work. So, better keep different "features" in different branches. For now, better make a branch for your work and keep this PR from master.

> No I am not rotating the already produced mesh.
I use ideasUnvToFoam Fluid_xy_original.unv to create the polyMesh directory.

I had not heard about this tool before. Interesting but maybe not really needed. I will check again in a second round.

> I tried rotating the mesh using the commands available in OpenFOAM, it works fine, but the rotation of CalculiX is a bit tricky and it does not exactly match with Fluid mesh. This is why I chose to make a new mesh all together.

What do you mean "does not exactly match with Fluid mesh"? Unless there is an offset/rotation, this should not be an issue for preCICE.

Ideally, we would like a rotated blockMeshDict and a rotated CalculiX configuration. This should be the cleanest solution and maybe not so difficult in this case, since the geometries are rather simple. I guess we should at least try to remove the Salome dependency and see what we can do afterwards.

MakisH commented:
We forgot about this for a while, but it is also fine, as https://github.com/precice/tutorials/pull/146 now rotated the case as part of a larger restructuring project.
